### PR TITLE
refactor: remove wtoken handler

### PR DESCRIPTION
**Pull request by 0xsambugs**
We are now deleting both `DCAHubCompanionParameters` and `DCAHubCompanionWTokenPositionHandler` (LOOK AT ALL THE DELETED CODE!!!)

The idea is to implement swap functionality by using the SwapAdapter or some of the extensions. So the wToken conversion will be just another swap

### CODE REVIEW

File: contracts/DCAHubCompanion/DCAHubCompanionLibrariesHandler.sol [hunk 1]
```diff
@@ -3,9 +3,9 @@ pragma solidity >=0.8.7 <0.9.0;
 
 import '../libraries/InputBuilding.sol';
 import '../libraries/SecondsUntilNextSwap.sol';
-import './DCAHubCompanionParameters.sol';
+import '../interfaces/IDCAHubCompanion.sol';
 
-abstract contract DCAHubCompanionLibrariesHandler is DCAHubCompanionParameters, IDCAHubCompanionLibrariesHandler {
```
Comment (0xsambugs) [inline contracts/DCAHubCompanion/DCAHubCompanionLibrariesHandler.sol:8]: This wasn't necessary before, not sure why it was there

File: contracts/interfaces/IDCAHubCompanion.sol [hunk 2]
```diff
@@ -3,226 +3,8 @@ pragma solidity >=0.8.7 <0.9.0;
...
-  /// @return The protocol token
-  // solhint-disable-next-line func-name-mixedcase
-  function PROTOCOL_TOKEN() external view returns (address);
-
-  /// @notice Returns the permission manager contract
-  /// @return The contract itself
-  function permissionManager() external view returns (IDCAPermissionManager);
-}
-
-interface IDCAHubCompanionSwapHandler is IDCAHubSwapCallee {
```
Comment (0xsambugs) [inline contracts/interfaces/IDCAHubCompanion.sol:32]: We weren't using this. I think we forgot to remove it when we moved the swapper to a new contract

File: test/integration/DCAHubCompanion/multi-call.spec.ts [hunk 3]
```diff
@@ -88,189 +88,189 @@ contract('Multicall', () => {
...
-        expect(balance).to.equal(hubWTokenBalanceAfterSwap.sub(swappedBalance));
-      });
-      then(`recipients's protocol balance increases`, async () => {
-        const balance = await ethers.provider.getBalance(recipient.address);
-        expect(balance).to.equal(initialRecipientProtocolBalance.add(swappedBalance));
-      });
-      thenCompanionRemainsWithoutAnyBalance();
-    });
-  });
+  // describe('protocol token as "from"', () => {
```
Comment (0xsambugs) [inline test/integration/DCAHubCompanion/multi-call.spec.ts:91]: Will un-comment when we implement the new swap functionality

File: test/unit/DCAHubCompanion/dca-hub-companion-hub-proxy-handler.spec.ts [hunk 4]
```diff
@@ -13,7 +13,6 @@ import {
 import { FakeContract, smock } from '@defi-wonderland/smock';
 import { TransactionResponse } from '@ethersproject/abstract-provider';
```
Comment (0xsambugs) [inline test/unit/DCAHubCompanion/dca-hub-companion-hub-proxy-handler.spec.ts:14]: Had forgotten to rename the test in the previous PR 😓

**Review by 0xged — APPROVED** [review]
NOICE! Looking good! I LOVE DELETING CODE 😍  ❤️‍🔥

**Review by nchamo — APPROVED** [review]
Damn, look at all that deleted code!!!

**Pull request merged**